### DISCUSSION TITLE
Provisioning schema in iModel

### DISCUSSION THREAD

**Discussion by ashish-srivastava1** [iTwin/itwinjs-core#2229] [Q&A] [answered]
How do I know whether iModel is provisioned with a particular domain schema from frontend code? If the schema is already loaded then I don't want to upload it again. The app has a UI control that would enable schema upload (happening in the backend) or not enable.

**Comment by pmconne** [comment, marked as answer]
There are currently virtually no frontend TypeScript APIs for querying information about the schemas stored in an iModel.
However, you should be able to query any metadata (including the list of all provisioned schemas, their versions, etc) by using [IModelConnection.query](https://www.itwinjs.org/reference/imodeljs-frontend/imodelconnection/imodelconnection/query/) to execute [meta queries](https://www.itwinjs.org/learning/ecsqltutorial/metaqueries/).
